waymo_raw_data_to_argoverse.py

```diff
@@ -30,205 +40,398 @@
+    """ """
+    if not Path(dirpath).exists():
+        os.makedirs(dirpath, exist_ok=True)
 
 
-def form_calibration_json(calib_data):
-	"""
-	Argoverse expects to receive "egovehicle_T_camera", i.e. from camera -> egovehicle, with
-		rotation parameterized as quaternion.
-	Waymo provides the same SE(3) transformation, but with rotation parmaeterized as 3x3 matrix
-	"""
-	calib_dict = {
-		'camera_data_': []
-	}
-	for camera_calib in calib_data:
-
-		cam_name = CAMERA_NAMES[camera_calib.name]
-		# They provide "Camera frame to vehicle frame."
-		# https://github.com/waymo-research/waymo-open-dataset/blob/master/waymo_open_dataset/dataset.proto
-		egovehicle_SE3_camera = np.array(camera_calib.extrinsic.transform).reshape(4,4)
-		x, y, z = egovehicle_SE3_camera[:3,3]
-		egovehicle_R_camera = egovehicle_SE3_camera[:3,:3]
-
-		assert np.allclose( egovehicle_SE3_camera[3], np.array([0,0,0,1]) )
-		egovehicle_q_camera = rotmat2quat(egovehicle_R_camera)
-		qw, qx, qy, qz = egovehicle_q_camera
-		f_u, f_v, c_u, c_v, k1, k2, p1, p2, k3 = camera_calib.intrinsic
-
-		cam_dict = {
-			'key': 'image_raw_' + cam_name,
-			'value': {
-				'focal_length_x_px_': f_u,
-				'focal_length_y_px_': f_v,
-				'focal_center_x_px_': c_u,
-				'focal_center_y_px_': c_v,
-				'skew_': 0,
-				'distortion_coefficients_': [0,0,0],
-				'vehicle_SE3_camera_': {
-					'rotation': {'coefficients': [qw, qx, qy, qz] },
-					'translation': [x,y,z]
-				}
-			}
-		}
-		calib_dict['camera_data_'] += [cam_dict]
-
-	return calib_dict
-
-
-def dump_pose(city_SE3_egovehicle, timestamp, log_id):
-	""" """
-	x,y,z = city_SE3_egovehicle[:3,3]
-	R = city_SE3_egovehicle[:3,:3]
-	assert np.allclose( city_SE3_egovehicle[3], np.array([0,0,0,1]) )
-	q = rotmat2quat(R)
-	w, x, y, z = q
-	pose_dict = {
-		'rotation': [w, x, y, z],
-		'translation': [x,y,z]
-	}
-	json_fpath = f'TEST_RAW_DATA/pose_logs/{log_id}/poses/city_SE3_egovehicle_{timestamp}.json'
-	check_mkdir(str(Path(json_fpath).parent))
-	save_json_dict(json_fpath, pose_dict)
-
-
-
-if __name__ == '__main__':
-
-	save_images = True
-	save_poses = True
-	save_calibration = True
-
-	main(save_images, save_poses, save_calibration)
+def save_json_dict(
+    json_fpath: Union[str, "os.PathLike[str]"], dictionary: Dict[Any, Any]
+) -> None:
+    """Save a Python dictionary to a JSON file.
 
+    Args:
+    json_fpath: Path to file to create.
+    dictionary: Python dictionary to be serialized.
+    """
+    with open(json_fpath, "w") as f:
+        json.dump(dictionary, f)
 
 
+def get_log_id_from_files(record_dir: str) -> List[str]:
+    """Get the log IDs of the Waymo records from the directory where they are stored
+
+    Args:
+        record_dir: The path to the directory where the Waymo data is stored
+    Returns:
+        log_ids: A list of log IDs from the Waymo dataset
+    """
+    files = glob.glob(f"{record_dir}/*.tfrecord")
+    log_ids = []
+    for file in files:
+        file = file.strip(record_dir)
```

Comment:
`strip()` can do some weird things, like 

```
>>> mystr = '/hello/world/its/me'
>>> mystr.strip('/hello/world/')
'its/m'
>>> mystr
'/hello/world/its/me'
>>> mystr.strip('/hello/world/its/')
'm'
```
Would `file.replace(record_dir, "")` be better?

Reply:
Addressed [here](https://github.com/johnwlambert/waymo_to_argoverse/pull/2/commits/9db2006d2741dbd66d59c22e07604a9bd62fa2a6)